share/wake/lib/core/vector.wake

```diff
@@ -61,15 +61,15 @@ export def vectorToList: Vector a => List a =
 #
 #   vempty (vseq 4) = False
 #   vempty (vseq 0) = True
-export def vempty (Vector _ s e): Boolean =
+export def vempty (Vector _ s e: Vector a): Boolean =
```

Comment:
I'm uncertain that a type signature of the same name as the destructor actually improves readability.

Reply:
Maybe not, and this might indeed be one place where we explicitly go against the style guide.  The one thing it does emphasize is that the function can take a `Vector` *of any type*.

Reply:
Actually, probably what would be more readable would be:
```
def vempty (v: Vector a): Boolean =
  def Vector _ s e = v
  ...
```

That function was written before we had the def extractor syntax available.